src/transport/websocket/stream.rs

```diff
@@ -124,7 +126,7 @@ impl<S: AsyncRead + AsyncWrite + Unpin> futures::AsyncWrite for BufferedStream<S
                         Poll::Ready(Err(_error)) =>
                             return Poll::Ready(Err(std::io::ErrorKind::UnexpectedEof.into())),
                         Poll::Pending => {
-                            self.state = State::ReadyToSend;
+                            self.state = State::FlushPending;
```

Comment:
Out of curiosity, how did you spot this?

Reply:
Went on a rabbit hole during this flaky test: https://github.com/paritytech/litep2p/pull/373#issuecomment-2804403610 🙈